framework/rest/IndexAction.php

```diff
@@ -54,6 +78,10 @@ protected function prepareDataProvider()
             return call_user_func($this->prepareDataProvider, $this);
         }
 
+        if ($this->searchClass instanceof SearchInterface) {
+            return (new $this->searchClass)->search(Yii::$app->request->queryParams);
+        }
```

Comment:
if you mean https://github.com/yiisoft/yii2/issues/4479#issuecomment-248318709 then I agree. It is much simpler. I think it just need to be documented somewhere.